etna/models/nn/deepstate/linear_dynamic_system.py

```diff
@@ -0,0 +1,226 @@
+        prior_cov:
+            Prior covariance matrix for latent state with shape (batch_size, latent_dim, latent_dim)
+        offset:
+            Offset for the target with shape (batch_size, seq_length, 1)
+        seq_length:
+            Length of the sequence.
+        latent_dim:
+            Dimension of the latent space.
+        """
+        self.emission_coeff = emission_coeff
+        self.transition_coeff = transition_coeff
+        self.innovation_coeff = innovation_coeff
+        self.noise_std = noise_std
+        self.prior_mean = prior_mean
+        self.prior_cov = prior_cov
+        self.offset = offset
+        self.seq_length = seq_length
+        self.latent_dim = latent_dim
+
+        self.batch_size = self.prior_mean.shape[0]
+        self._eye = torch.eye(self.latent_dim).type_as(noise_std)
+
+    def kalman_filter_step(
+        self,
+        target: Tensor,  # (batch_size, 1)
+        noise_std: Tensor,  # (batch_size, 1)
+        prior_mean: Tensor,  # (batch_size, latent_dim)
+        prior_cov: Tensor,  # (batch_size, latent_dim, latent_dim)
+        emission_coeff: Tensor,  # (batch_size, latent_dim)
+        offset: Tensor,  # (batch_size, 1)
+    ):
+        """One step of the Kalman filter.
+
+        This function computes the filtered state (mean and covariance) given the
+        LDS coefficients in the prior state (mean and variance) and observations.
+
+        Parameters
+        ----------
+        target:
+            Observations of the system with shape (batch_size, 1)
+        noise_std:
+            Standard deviation of the observations noise with shape (batch_size, 1)
+        prior_mean:
+            Prior mean of the latent state with shape (batch_size, latent_dim)
+        prior_cov:
+            Prior covariance of the latent state with shape (batch_size, latent_dim, latent_dim)
+        emission_coeff:
+            Emission coefficient with shape (batch_size, latent_dim)
+        offset:
+            Offset for the observations with shape (batch_size, 1)
+
+        Returns
+        -------
+        :
+            Log probability with shape (batch_size, 1)
+        :
+            Filtered_mean with shape (batch_size, latent_dim, 1)
+        :
+            Filtered_covariance with shape (batch_size, latent_dim, latent_dim)
+        """
+        emission_coeff = emission_coeff.unsqueeze(-1)
+
+        # H * mu (batch_size, 1)
+        target_mean = (emission_coeff.permute(0, 2, 1) @ prior_mean.unsqueeze(-1)).squeeze(-1)
+        # v (batch_size, 1)
+        residual = target - target_mean - offset
+        # R (batch_size, 1, 1)
+        noise_cov = torch.diag_embed(noise_std * noise_std)
+        # F (batch_size, 1, 1)
+        target_cov = emission_coeff.permute(0, 2, 1) @ prior_cov @ emission_coeff + noise_cov
+        # K (batch_size, latent_dim)
+        kalman_gain = (prior_cov @ emission_coeff @ torch.inverse(target_cov)).squeeze(-1)
```

Comment:
Actually `inverse` is a dummy implementation -- it is not the best practice to implement it this way. May be you can search for better implementation(see gluons one)

Reply:
`target_cov` is (batch_size, 1, 1). We can replace `torch.inverse(target_cov)` with `(1 / target_cov)`.